packages/salesforcedx-vscode-soql/test/vscode-integration/client/client.test.ts

```diff
@@ -69,16 +61,57 @@ describe('SOQL language client', () => {
     expect(diagnostics[0].message).to.equal(`missing 'from' at 'Account'`);
   });
 
-  it('should create diagnostics based off of limit 0 execute error results', async () => {
-    const expectedError = `SELECT Ids FROM ACCOUNT\nERROR at Row:1:Column:8\nSome error at 'Ids'`;
-    await workspace.fs.writeFile(
-      soqlFileUri,
-      encoder.encode(`
-      SELECT Ids
-      FROM Account
-    `)
+  it('should not create diagnostics based off of remote query validation by default', async () => {
+    soqlFileUri = await writeSOQLFile(
+      'testSemanticErrors_remoteRunDefault.soql',
+      'SELECT Ids FROM Account'
     );
 
+    const querySpy = sandbox.stub(mockConnection, 'query');
+    await window.showTextDocument(soqlFileUri);
+
+    await sleep(100);
+
+    expect(querySpy.notCalled).to.be.true;
+    const diagnostics = languages.getDiagnostics(soqlFileUri);
+    expect(diagnostics)
+      .to.be.an('array')
+      .to.have.lengthOf(0);
+  });
+
+  it('should not create diagnostics based off of remote query validation when disabled', async () => {
+    soqlFileUri = await writeSOQLFile(
+      'testSemanticErrors_remoteRunDefault.soql',
+      'SELECT Ids FROM Account'
+    );
+
+    stubSOQLExtensionConfiguration(sandbox, {
+      'experimental.soqlEditorRemoteChecks': true
```

Comment:
I think this test needs to be fixed.  'experimental.soqlEditorRemoteChecks': true and the test is passing.

1) The test should be fixed so that the mock connection throws an error
2) experimental.soqlEditorRemoteChecks should be set to false

Reply:
Great catch!!!  Fixed now. Very tricky:

Seems like the integration test framework runs a failed test multiple times.  And in the case of language diagnostics, the second time it run for the exact same file and input, no diagnostics check happens (I guess this is an optimization on VSCode. That's also how it haves when testing manually).

Here's what happened:
1. Assertion `expect(querySpy.notCalled).to.be.true` failed
2. Test re-run
3. VSCode didn't run the diagnostic check again (optimization)
4. Thus, assertion `expect(querySpy.notCalled).to.be.true` and the rest passed because no validation happened

Thanks!